_posts/2023-09-11-mcu-peripheral-forwarding.md

```diff
@@ -0,0 +1,336 @@
+    
+    
+    ```cpp
+    /**
+     * @brief remcu_connect2OpenOCD
+     * The function is used to create a connection to the OpenOCD server destination
+     * @param host - ip address of OpenOCD server "XXX.XXX.XXX.XXX". If the server runs on the host machine, the argument should be "localhost" or "127.0.0.1" value.
+     * @param port - port of OpenOCD server. Default is 6666
+     * @param timeout_sec - This parameter specifies the length of time, in seconds, to wait for a response when establishing communications. It can not be negative!
+     * @return If no error occurs, function returns true
+     */
+    REMCULIB_DLL_API bool remcu_connect2OpenOCD(const char* host, const uint16_t port,
+                          const int timeout_sec);
+    
+    /**
+     * @brief remcu_connect2GDB
+     * The function is used to create a connection to the GDB server destination
+     * @param host - ip address of GDB server "XXX.XXX.XXX.XXX". If the server runs on the host machine, the argument should be "localhost" or "127.0.0.1" value.
+     * @param port - port of GDB server. Default of OpenOCD is 3333
+     * @param timeout_sec - This parameter specifies the length of time, in seconds, to wait for a response when establishing communications. It can not be negative!
+     * @return If no error occurs, function returns true
+     */
+    REMCULIB_DLL_API bool remcu_connect2GDB(const char* host, const uint16_t port,
+                           const int timeout_sec);
+    ```
+    
+3. Reset the chip: Once the connection is established, you can utilize the [remcu_resetRemoteUnit](https://github.com/remotemcu/remcu/blob/master/export/remcu.h#L107) function to reset the MCU. This step ensures a clean starting point and prepares the chip for further operations.
+    
+    
+    ```cpp
+    remcu_resetRemoteUnit(__HALT);
+    ```
+    
+
+Now, You can execute any SDK driver functions…
+
+
+## Conclusion
+
+In conclusion, we've explored the fundamentals of Chip Peripheral Forwarding technology, laying down the basis for its understanding and implementation.
+This approach emerges as a versatile technology, simplifying the expansion of peripherals for both PC and embedded systems like Raspberry Pi. Moreover, it significantly bolsters hardware equipment testing, harnessing the capabilities of PC program tools such as high-level languages like Python and interactive shells like Jupyter notebook and it eliminates the need for firmware and communication protocol development.
+
+
+To witness this technology in action, one need look no further than the [**REMCU examples repository**](https://github.com/remotemcu/remcu_examples). Here, a diverse array of practical examples is offered, specifically crafted for various MCUs.
+
+
+
+![https://github.com/remotemcu/remcu_examples/raw/master/stm32f4_discovery/accell_graph/img/mems_demo.gif](https://github.com/remotemcu/remcu_examples/raw/master/stm32f4_discovery/accell_graph/img/mems_demo.gif)
```

Comment:
Could you add more details in the above text about the gif? Is this example leveraging REMCU?

Reply:
Certainly, the example does leverage REMCU. I've updated the text to include a more detailed description of how REMCU is used in the provided GIF. Thank you for your suggestion!